### ISSUE TITLE
Logits Processors

### ISSUE THREAD

**Issue by psitronic** [microsoft/DeepSpeed-MII#532]
Is there a way to add functions that modify logits based on previously generated tokens? E.g., to apply an output formatter. Something similar to a `logits_processors` argument in e.g. vLLM.
